encoding/descaling.md

```diff
@@ -0,0 +1,356 @@
+without dabbling into specific masks.
+However, so long as you've got
+a source with nice, clean lineart,
+descaling might be a viable option,
+and possibly nullify the need
+of a lot of other filtering.
+
+
+## Preparation
+
+To prepare for figuring out the native resolution,
+you'll want to use [getnative][],
+a Python script designed around
+figuring out the resolution
+a show was animated at.
+For the actual descaling,
+make sure to grab BluBb-mADe's [descale][].
+
+One important thing
+to keep in mind when descaling
+is that you will never find
+"the perfect descaling settings".
+This is because the best available sources to consumers,
+usually Blu-rays,
+aren't lossless.
+There's always going to be some differences
+from the original masters
+which makes it impossible
+to perfectly descale something.
+However, usually those differences
+are so small that they're negligible.
+If you run into a case where
+you can only get high relative errors,
+descaling can be highly destructive.
+It's instead recommended to downscale
+as you would normally,
+or to not mess with scaling at all.
+
+[getnative]: https://github.com/Infiziert90/getnative
+[descale]: https://github.com/BluBb-mADe/vapoursynth-descale
+
+
+## Finding out the native resolution
+
+To figure out what
+the native resolution of an anime is,
+first you need a good frame to test.
+Ideally you'll want a bright frame
+with as little blur as possible of high quality
+(Blu-ray or very good webstreams).
+It also helps to not have
+too many post-processed elements in the picture.
+Whilst it is most definitely possible
+to get pretty good results with "bad" frames,
+you'll want to generally lower that chance
+as much as possible.
+
+Here's some examples of "bad" frames.
+
+![Manaria Friends Ep01 frame 1](images/descale_manaria01.png)
+
+This picture is dark.
+It also has some effects over it.
+
+![Manaria Friends Ep01 frame 2](images/descale_manaria02.png)
+
+This picture is also very dark,
+and has even more effects over it.
+
+![Manaria Friends Ep01 frame 3](images/descale_manaria03.png)
+
+Heavy dynamic grain will almost always give bad results.
+
+![Manaria Friends Ep01 frame 4](images/descale_manaria04.png)
+
+This is a nice frame to use as reference.
+The background is a bit blurry,
+but it isn't full of effects
+and is fairly bright.
+The lineart is very clear.
+
+We will now make use of the getnative.py script
+to figure out what resolution
+this anime was produced at.
+Run the following in your terminal:
+
+```bash
+$ python getnative.py "Manaria Friends Ep01 frame 4.png"
+```
+
+It should show the following:
+
+```
+Using imwri as source filter
+501/501
+Kernel: bicubic AR: 1.78 B: 0.33 C: 0.33
+Native resolution(s) (best guess): 878p
+done in 18.39s
+```
+
+If you look in the directory
+that you ran the script in,
+you will find a new folder
+called "getnative".
+You can find the following graph
+in there as well:
+
+![Manaria_Friends-ep1_04_graph](images/descale_graph.png)
+
```

Comment:
This is also the place to discuss what are considered to be "low relative errors" and "high relative errors", when to ignore them, and how to interpret them and the graph.

Reply:
Expanded greatly on how to read the graphs, as well as showing some examples.